gateway-js/src/__tests__/build-query-plan.feature

```diff
@@ -1405,7 +1418,8 @@ Scenario: multiple root mutations with correct service order
             ],
             "variableUsages": [],
             "operationKind": "query",
-            "operation": "query($representations:[_Any!]!){_entities(representations:$representations){...on User{reviews{product{__typename ...on Book{__typename isbn}...on Furniture{upc}}}}}}"
+            "operation": "query LoginAndReview_reviews_2($representations:[_Any!]!){_entities(representations:$representations){...on User{reviews{product{__typename ...on Book{__typename isbn}...on Furniture{upc}}}}}}",
+            "operationName": "LoginAndReview_reviews_2"
```

Comment:
@pcmanus @lennyburdette Picking a random place to leave this post-merge thought — would it make sense to use double-underscores as the delimiter to create a more predictable pattern for parsing and to avoid name-conflicts across operation + subgraph names?  Mostly just because it seems not completely unlikely that both an operation and a subgraph might use `_` on its own and, while not forbidden, much less likely that someone would use `__`.   We could even forbid `__` as a valid graph name (and probably should or may already have a enforced character set somewhere?).

Just trying to avoid the case of two things end up being exactly equal.

It was also very intentional that I'd suggested prefixing with `__` just to indicate the generatedness and avoid clashes, just in case people use these operations downstream in meaningful ways.

Also happy to ticket this for follow-up.   Or not, depending on thoughts.

Reply:
no strong feelings either way. i went with fewer characters but four extra underscores (two in the operationName query param and two in the operation document) isn't a huge deal.

Reply:
I also like the idea of the double underscore separator, and if we are not already we should enforce that subgraphs have to start and end with a number or letter